src/main/minio.ts

```diff
@@ -941,23 +978,32 @@ export class Client {
   // * `objectName` _string_: name of the object
   // * `getOpts` _object_: Version of the object in the form `{versionId:'my-uuid'}`. Default is `{}`. (optional)
   // * `callback(err, stream)` _function_: callback is called with `err` in case of error. `stream` is the object content stream
-  getObject(bucketName, objectName, getOpts={}, cb) {
+  getObject(bucketName: string, objectName: string, callback: (err: Error | null) => void): void
+  getObject(bucketName: string, objectName: string, getOpts: GetObjectOptions, callback: (err: Error | null) => void): void
+  getObject(bucketName: string, objectName: string, ...rest: [(err: Error | null) => void] | [GetObjectOptions, (err: Error | null) => void]): void {
```

Comment:
This will be the style of pattern on how we'll implement function overloading for the sake of backward compatibility. See https://tsplay.dev/wXvnOm for Typescript playground sample.

Reply:
i will try to update types for  the above links. Thank you @aldy505 . 

Could you please resolve the conflicts ?